pandas/core/sorting.py

```diff
@@ -52,7 +52,17 @@ def _int64_cut_off(shape):
                 return i
         return len(shape)
 
-    def loop(labels, shape):
+    def maybe_lift(lab, size):  # pormote nan values
```

Comment:
sp here

can you add a 1-line comment on what this is doing

Reply:
Same as with loop - I would try, though I'm not the author of original code - I've just changed it from recursion to loop, so I can't be sure I understand 100% all the nuances here...